### DISCUSSION TITLE
Does Octokit support repository rulesets endpoints?

### DISCUSSION THREAD

**Discussion by muhammad-umair-sec** [octokit/octokit.rb#1713] [Q&A]
I am trying to access repository rulesets, but I am getting this error:
`The GitHub API does support working with repository rulesets, but it's not yet officially documented in the Octokit.rb gem`.
Does Octokit support this endpoint?

**Comment by timgentry** [comment]
Hi there, `client.get("repos/#{repo.full_name}/rules/branches/#{branch_name}")` works for me or `client.get("repos/#{repo.full_name}/rulesets")` if you want all the rules for a repo, rather than a specific branch. API details are [here](https://docs.github.com/en/rest/repos/rules?apiVersion=2022-11-28#get-rules-for-a-branch) and you can see the `get` command mirrors the API calls. I can't see any convenience methods wrapping those underlying calls for the moment. Cheers
